### ISSUE TITLE
add option to control devices?

### ISSUE THREAD

**Issue by vampywiz17** [maku2903/tuya-2-mqtt#1]
Hello there!

Your project are really cool. it do a similar project, but is also not plan to control devices. 

https://github.com/vampywiz17/tuyaiot2mqtt

Are you plan it?

**Comment by maku2903** [comment]
Hello, it would be a nice feature, I thought about it but I didn't plan to implement it till now. 
Frankly, I wrote this project to get readings of tuya thermometers into graphana dashboard because of urgent need.

~~What do you want control?~~
--edit--
_Wrong question.._

My quick sketch of pipeline of controlling device:
1. Loop Event listener on mqtt broker messages
2. Catch/filter from broker mqtt msgs control topic schema of controlling device f.e. `ctl/tuya/{{ device_id }}`
3. Get content of controll topic, that will be tuya command json f.e. 
`{
  "commands": [
    {
      "code": "switch_1",
      "value": true
    }
  ]
}
`
4. Pass it to tuya cloud api
5. Send to mqtt broker topic message on success or fail to control based on tuya api response

If this pipeline would meet you expectations i can deliver feature maybe even this week (I need to buy tuya device that is controllable for dev purposes, i have only basic tuya thermometers :^) )

--edit 2--
For local tuya control i would have to dive into tuya local api if its not proprietary. For now i dont even know how to locally establish connection with tuya.

**Comment by maku2903** [comment]
@vampywiz17 
If you want to test
RC1 image under [1.1.0-RC1](https://hub.docker.com/layers/makuuu2903/tuya-2-mqtt/1.1.0-RC1/images/sha256-f2d8d269f13081df2edbbf03eee8a12a1c8f7aab196bf663137a074d51ceefe8) or just use python code from dev branch 7fb7296

Send commands to mqtt on topic: `tele/tuya/cloud/<tuya_device_id>/command` where MQTT message is json list of commands f.e. `'[{"cmd_key1": "cmd_key1_value"}]'`

Example shell:
`mosquitto_pub -h <mqtt_host> -p <mqtt_port> -t tele/tuya/cloud/<tuya_device_id>/command -u <mqtt_user> -P <MQTT_pass> -m '[{"cmd_key1": "cmd_key1_value"}]'
`

**Message topic is hardcoded for now**

Probably works, i dont have controllable tuya devices.

**Comment by vampywiz17** [comment]
It seem not work. I try it co control one of my devices.

**Comment by maku2903** [comment]
I did testing, 100% sure problem related with [platform.tuya.com/cloud](platform.tuya.com/cloud) account type/settings/internal tuya settings.
It turned out that my thermometers are controllable (i can set maxtemp maxhumidity etc.) so i begun testing.

I got errors on responses for request tinytuya.Cloud().sendcommand.
Error log of response from tuya cloud:
`Command sent to Tuya: xxxxx -> {'commands': [{'maxtemp_set': 600}]}, response: {'code': 500, 'msg': 'system error, please contact the admin', 'success': False, 't': yyyyy, 'tid': 'zzzzz'}`

Unfortunately, tid wont ever bring any error documentation from tuya error api. Tried many times.

In [tuya api explorer](https://platform.tuya.com/cloud/explorer) sendcommand method works somehow.

According to error code = 500 and [this](https://developer.tuya.com/en/docs/iot/error-code?id=K989ruxx88swc) i have to place ticket [here](https://service.console.tuya.com/8/3/list?source=support_center)

Summing up: app should work if tuya api will work...

**Comment by vampywiz17** [comment]
Check my dev branch. I hope it Help!

https://github.com/vampywiz17/tuyaiot2mqtt

**Comment by maku2903** [comment]
tuya support replied that devices are offline - when device is offline then API responds error 500. Unfortunately , thats positive: tuya thermometers are activating only themselves 1 time / 1 hour, so they cant receive command messages while in sleep mode. Need get another controllable tuya thing that can be online 24/7 for debuging.
